pkg/util/winutil/pdhutil/pdh_386.go

```diff
@@ -22,7 +22,6 @@ type PDH_FMT_COUNTERVALUE_LARGE struct {
 
 // Union specialization for long values
 type PDH_FMT_COUNTERVALUE_LONG struct {
-	CStatus      uint32
-	floatpadding uint32
-	LongValue    int32
```

Comment:
Are you sure about this change? 
`DoubleValue`, `LargeValue` and `LongValue` offset on 32 bits is also 8 bytes.

The following lines display 3 times `8` for x86
```c++
	std::cout << offsetof(_PDH_FMT_COUNTERVALUE, longValue) << std::endl;			
	std::cout << offsetof(_PDH_FMT_COUNTERVALUE, doubleValue) << std::endl;
	std::cout << offsetof(_PDH_FMT_COUNTERVALUE, largeValue) << std::endl;
```

Reply:
I don't understand.  The _386 code has no padding.

Reply:
From pkg/util/winutil/pdhutil/pdh_386.go
In `PDH_FMT_COUNTERVALUE_DOUBLE` the offfset of `DoubleValue` field is 8.
In `PDH_FMT_COUNTERVALUE_LARGE ` the offfset of `LargeValue` field is 8.
In `PDH_FMT_COUNTERVALUE_LONG ` the offfset of `LongValue` field is **4** (and not 8 as it is the case for the other fields). 

If you compute the offset of these fields in C++, it is always 8 (even for `LongValue`) both on `x86` and `x64`.

From my understanding, you might need to have` floatpadding` in `PDH_FMT_COUNTERVALUE_LONG` (In the same way as for `PDH_FMT_COUNTERVALUE_DOUBLE` and `PDH_FMT_COUNTERVALUE_LARGE`).

```go
// In pkg/util/winutil/pdhutil/pdh_386.go
type PDH_FMT_COUNTERVALUE_LONG struct {
	CStatus   uint32
        floatpadding uint32 . // <---- Add this line 
	LongValue int32
}
```